tests/otel_integrations/test_mysql.py

```diff
@@ -0,0 +1,113 @@
+import mysql.connector
+import pytest
+from dirty_equals import IsInt
+from inline_snapshot import snapshot
+from testcontainers.mysql import MySqlContainer
+
+import logfire
+from logfire.testing import TestExporter
+
+
+@pytest.fixture(scope='module')
+def mysql_container():
+    with MySqlContainer() as mysql_container:
+        yield mysql_container
+
+
+def get_mysql_connection(mysql_container: MySqlContainer):
+    host = mysql_container.get_container_host_ip()
+    port = mysql_container.get_exposed_port(3306)
+    connection = mysql.connector.connect(host=host, port=port, user='test', password='test', database='test')
+    return connection
+
+
+def test_mysql_instrumentation(exporter: TestExporter, mysql_container: MySqlContainer):
+    logfire.instrument_mysql()
+    conn = get_mysql_connection(mysql_container)
+    cursor = conn.cursor()
+    cursor.execute('DROP TABLE IF EXISTS test')
+    cursor.execute('CREATE TABLE test (id INT PRIMARY KEY, name VARCHAR(255))')
+    assert exporter.exported_spans_as_dict() == snapshot(
+        [
+            {
+                'name': 'DROP',
+                'context': {'trace_id': 1, 'span_id': 1, 'is_remote': False},
+                'parent': None,
+                'start_time': 1000000000,
+                'end_time': 2000000000,
+                'attributes': {
+                    'logfire.span_type': 'span',
+                    'logfire.msg': 'DROP TABLE IF EXISTS test',
+                    'db.system': 'mysql',
+                    'db.name': 'test',
+                    'db.statement': 'DROP TABLE IF EXISTS test',
+                    'db.user': 'test',
+                    'net.peer.name': 'localhost',
+                    'net.peer.port': IsInt(),
+                },
+            },
+            {
+                'name': 'CREATE',
+                'context': {'trace_id': 2, 'span_id': 3, 'is_remote': False},
+                'parent': None,
+                'start_time': 3000000000,
+                'end_time': 4000000000,
+                'attributes': {
+                    'logfire.span_type': 'span',
+                    'logfire.msg': 'CREATE TABLE test (id INT PRIMARY KEY, name VARCHAR(255))',
+                    'db.system': 'mysql',
+                    'db.name': 'test',
+                    'db.statement': 'CREATE TABLE test (id INT PRIMARY KEY, name VARCHAR(255))',
+                    'db.user': 'test',
+                    'net.peer.name': 'localhost',
+                    'net.peer.port': IsInt(),
+                },
+            },
+        ]
+    )
+
+
+def test_instrument_mysql_connection(exporter: TestExporter, mysql_container: MySqlContainer):
+    conn = get_mysql_connection(mysql_container)
+    conn = logfire.instrument_mysql(conn)  # type: ignore
```

Comment:
Why the `type: ignore`? We don't want users to have to write that.

Please extend this test to show that only this specific connection is being instrumented. You will probably have to uninstrument the module at the end of `test_mysql_instrumentation` to ensure that another uninstrumented connection doesn't create spans.

Reply:
`type: ignore` was added because `get_mysql_connection` return type was not mentioned. If we add `MySQLConnection` as return type then we are still getting type errors. Fixed it with `Any`

Reply:
This test still needs to show that an uninstrumented connection doesn't create spans.